apps/data/lib/data/opportunities.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Data.Opportunities do
+  @moduledoc """
+  The public API for `opportunity` resources
+  """
+
+  alias Data.{Opportunity, Project, Repo}
+
+  def get(params), do: Repo.get_by(Opportunity, params)
+
+  def insert(params) do
+    %Opportunity{}
+    |> Opportunity.changeset(params)
+    |> Repo.insert
+  end
+
+  def update(id, params) when is_integer(id) do
+    Opportunity
+    |> Repo.get(id)
+    |> update(params)
+  end
+
+  def update(struct, params) do
```

Comment:
Add a guard for this one too? I think its worth adding some simple docstrings for all these maybe, too. Doesnt have to be fancy or in depth but a doctest for each wouldnt hurt and could keep things clean and easy to use.